src/main/java/com/contentgrid/gateway/collections/ObservableMap.java

```diff
@@ -32,6 +31,11 @@ public V put(K key, V value) {
         return previousValue;
     }
 
+    @Override
+    public Set<K> keySet() {
+        return this.map.keySet();
+    }
```

Comment:
Why can't you just keep delegating that?

Reply:
`@Delegate` and generic types have issues